plugins/node/opentelemetry-instrumentation-mysql/src/types.ts

```diff
@@ -16,4 +16,17 @@
 
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
 
-export type MySQLInstrumentationConfig = InstrumentationConfig;
+/**
+ * Function that can be used to serialize db.statement tag
+ * @param statement - mysql statement object
+ *
+ * @returns serialized string that will be used as the db.statement attribute.
+ */
+export type DbStatementSerializer = (statement: string) => string;
+
+export interface MySQLInstrumentationConfig extends InstrumentationConfig {
+  /**
+   * Custom serializer function for the db.statement tag
+   */
+  dbStatementSerializer?: DbStatementSerializer;
```

Comment:
`dbStatementSerializer` is used in redis and mongo instrumentations where the db.statement needs to be constructed from object to text (hence the "Serializer").

For SQL instrumentation, the statement is already present as an SQL string, and the convention is to use `enhancedDatabaseReporting` which is a boolean that indicates whether to collect the SQL (potentially sensitive) "values" or not (not sure it's consistent but that is the general attitude)

I suggest considering aligning with these conventions